src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java

```diff
@@ -106,7 +106,7 @@
         Stream.of(
             TokenTypes.IDENT, TokenTypes.STRING_LITERAL, TokenTypes.CHAR_LITERAL,
             TokenTypes.NUM_LONG, TokenTypes.NUM_INT, TokenTypes.NUM_DOUBLE, TokenTypes.NUM_FLOAT,
-            TokenTypes.TEXT_BLOCK_CONTENT)
+            TokenTypes.TEXT_BLOCK_CONTENT, TokenTypes.COMMENT_CONTENT)
```

Comment:
Please confirm and show example(s). With your PR, am I able to suppress a violation by comment with xpath suppression if the check producing the violation is not comment aware and is the only check in the config? If the answer is no, if a comment aware check is add to the same config, does it then work?

@romani ping

Reply:
https://checkstyle.org/config_coding.html#MatchXpath is also not a comment aware check, so it can't support comments with this fix.

I believe this has been an issue that only comment aware checks can benefit from xpath'ing on a comment.

Reply:
Found the issue https://github.com/checkstyle/checkstyle/issues/7531

Reply:
Is this issue related to this PR. This seems to be an independent issue. Please share your views.

Reply:
It's independent, I just want confirmation that suppressing with a comment will only work with a comment aware check and will fail with a non-comment aware check. See first post in this chain.

Reply:
this is not related to this PR problems, we will address them separately.

Reply:
Now I see why this point was raised, it is confusing :).
Please move doc update to TrailingComment Check, as this feature is for it. 
Filter does not know logic and nuances of any Check implementation.

Reply:
Done, I had a similar thought as I said in https://github.com/checkstyle/checkstyle/pull/9832#discussion_r610012192